### ISSUE TITLE
[BUG] swanlab.config can't fit hydra - is not JSON serializable

### ISSUE THREAD

**Issue by xiaolin199912** [SwanHubX/SwanLab#237]
## 🤩 Features description [Please make everyone to understand it]

test code: 
```python
import hydra
from omegaconf import DictConfig, OmegaConf
import json
import swanlab


@hydra.main(config_path=".", config_name="configs", version_base="1.3.2")
def startup(cfg: DictConfig):
    config = dict(cfg)
    print(config)
    print("#" * 30)
    swanlab.init(experiment_name="abc", config=config)
   
startup()
```

configs.yaml, locate in the same dir:

```yaml
data:
  dir_image: "./data/samimage256/256/"
  dir_mask: "./data/samimage256/mymask/"
  image_size: 512

hardware:
  seed: 2021
  num_workers: 4
```

Error:
```console
Traceback (most recent call last):
  File "/Users/zeyilin/miniforge3/envs/test_py310/lib/python3.10/site-packages/swanlab/data/run/main.py", line 310, in __check_config
    ujson.dumps(config)
TypeError: {'dir_image': './data/samimage256/256/', 'dir_mask': './data/samimage256/mymask/', 'image_size': 512} is not JSON serializable
```

**Comment by xiaolin199912** [comment]
If use following code, run will normal:

```python
import hydra
from omegaconf import DictConfig, OmegaConf
import json
import swanlab


@hydra.main(config_path=".", config_name="configs", version_base="1.3.2")
def startup(cfg: DictConfig):
    config = OmegaConf.to_container(cfg, resolve=True)
    json.dumps(config)
    print(config)
    print("#" * 30)
    swanlab.init(experiment_name="abc", config=dict(cfg))

startup()
```

**Comment by xiaolin199912** [comment]
Fix: #238 
Use a json preprocess function to fix this problem.
Maybe in the future, we should test more framework'config method to make the preprocess method better 😄
